middleware.ts

```diff
@@ -18,6 +18,10 @@ export function middleware(req: NextRequest) {
 
   if (!hostname) return
 
+  if(hostname==`${process.env.NEXT_PUBLIC_APP_DOMAIN}` && (path=="/signin" || path=="/signup")){ 
+    return NextResponse.redirect(`http://app.${process.env.NEXT_PUBLIC_APP_DOMAIN}${path}`);
+  }
```

Comment:
Changes Required
1. Take the `app` using process.env.NEXT_PUBLIC_APP_DASHBOARD_DOMAIN
2. Take the `https` or `http` based on the incoming request protocol.

Reply:
@piyushgarg-dev, could you review this PR now, I made the changes required

Reply:
@Hari1122-code 
Prev solution was valid where you were checking /signin and /signup routes. We don't want to redirect every route to app subdomain.

Reply:
Yes, you are correct @piyushgarg-dev  and sorry for the extra commits now I have again resolved the changes.
Please review it